src/routes/blog/index.json.ts

```diff
@@ -1,9 +1,22 @@
+import { readFileSync } from 'fs';
 import type { RequestHandlerOutput } from '@sveltejs/kit/types/internal';
 import { fetchNewsletters } from './_query';
+import { DEBUG } from '$lib/constants';
+
+const SAMPLE_POST = readFileSync('./src/routes/blog/_testdata/posts.json');
 
 export async function get(): Promise<RequestHandlerOutput> {
-  return new Response(JSON.stringify(await fetchNewsletters()), {
-    status: 200,
-    headers: { 'Content-Type': 'application/json' },
-  });
+  // Uses sample data when env variables are not satisfied.
+  const isSatisfied =
+    import.meta.env.VITE_GH_ACCESS_TOKEN === undefined ||
+    import.meta.env.VITE_GH_DISCUSSION_CATEGORY_ID === undefined;
+
+  // remove the ! infront of the DEBUG to show only the sample posts
+  return new Response(
+    !DEBUG || isSatisfied ? SAMPLE_POST : JSON.stringify(await fetchNewsletters()),
+    {
+      status: 200,
+      headers: { 'Content-Type': 'application/json' },
+    }
+  );
```

Comment:
I changed the conditions slightly to make this endpoint always behave as expected: when debug mode is on or the required environment variables are unsatisfied, we will always use the sample data. Otherwise, when debug mode is off and all of the required environment variables are populated, the server will attempt to fetch the latest blog data using `fetchNewsletters()`. 
```suggestion
  // Uses sample data when DEBUG=1 or env variables are not satisfied.
  const isSatisfied =
    import.meta.env.VITE_GH_ACCESS_TOKEN !== undefined &&
    import.meta.env.VITE_GH_DISCUSSION_CATEGORY_ID !== undefined;

  return new Response(
    DEBUG || !isSatisfied ? SAMPLE_POST : JSON.stringify(await fetchNewsletters()),
    {
      status: 200,
      headers: { 'Content-Type': 'application/json' },
    }
  );
```

Reply:
Done